blockchain-albatross/src/blockchain.rs

```diff
@@ -1239,72 +1220,56 @@ impl Blockchain {
             &read_txn
         };
 
+        // Gets slots collection from either the cached ones, or from the macro block.
         let slots_owned;
-        let slots;
-        if policy::epoch_at(state.block_number()) == policy::epoch_at(block_number) {
-            slots = state.current_slots.as_ref().expect("Missing current epoch's slots");
-        } else if policy::epoch_at(state.block_number()) == policy::epoch_at(block_number) + 1 {
-            slots = state.last_slots.as_ref().expect("Missing previous epoch's slots");
-        } else {
+        let slots = if policy::epoch_at(state.block_number()) == policy::epoch_at(block_number) {
+            state.current_slots.as_ref().expect("Missing current epoch's slots")
+        }
+        else if policy::epoch_at(state.block_number()) == policy::epoch_at(block_number) + 1 {
+            state.last_slots.as_ref()
+                .unwrap_or_else(|| panic!("Missing previous epoch's slots for block {}.{}", block_number, view_number))
+        }
+        else {
             let macro_block = self.chain_store
                 .get_block_at(policy::macro_block_before(block_number), true, Some(&txn))
-                .or_else(|| {
-                    warn!("Failed to determine slots - preceding macro block not found: block_number={}, view_number={}, state.block_number()={}", block_number, view_number, state.block_number());
-                    None
-                })?
+                .unwrap_or_else(|| panic!("Failed to determine slots - preceding macro block not found: block_number={}, view_number={}, state.block_number()={}", block_number, view_number, state.block_number()))
```

Comment:
Actually, this *can happen* once we implemented a fast sync method based on zk-proofs!